crates/fj-core/src/validate/sketch.rs

```diff
@@ -129,4 +172,79 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn should_find_clockwise_exterior_cycle() -> anyhow::Result<()> {
+        let mut core = Core::new();
+
+        let valid_outer_circle =
+            HalfEdge::circle([0., 0.], 1., &mut core).insert(&mut core);
+        let valid_exterior =
+            Cycle::new(vec![valid_outer_circle.clone()]).insert(&mut core);
+        let valid_sketch =
+            Sketch::new(vec![
+                Region::new(valid_exterior.clone(), vec![]).insert(&mut core)
+            ]);
+        valid_sketch.validate_and_return_first_error()?;
+
+        let invalid_outer_circle = HalfEdge::from_sibling(
+            &valid_outer_circle,
+            Vertex::new().insert(&mut core),
+        )
+        .insert(&mut core);
+        let invalid_exterior =
+            Cycle::new(vec![invalid_outer_circle.clone()]).insert(&mut core);
+        let invalid_sketch =
+            Sketch::new(vec![
+                Region::new(invalid_exterior.clone(), vec![]).insert(&mut core)
+            ]);
+        assert_contains_err!(
+            invalid_sketch,
+            ValidationError::Sketch(
+                SketchValidationError::ClockwiseExteriorCycle()
+            )
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn should_find_counterclockwise_interior_cycle() -> anyhow::Result<()> {
+        let mut core = Core::new();
+
+        let outer_circle =
+            HalfEdge::circle([0., 0.], 2., &mut core).insert(&mut core);
+        let inner_circle =
+            HalfEdge::circle([0., 0.], 1., &mut core).insert(&mut core);
+        let cw_inner_circle = HalfEdge::from_sibling(
+            &inner_circle,
+            Vertex::new().insert(&mut core),
+        )
+        .insert(&mut core);
+        let exterior = Cycle::new(vec![outer_circle.clone()]).insert(&mut core);
+
+        let valid_interior =
+            Cycle::new(vec![cw_inner_circle.clone()]).insert(&mut core);
+        let valid_sketch = Sketch::new(vec![Region::new(
+            exterior.clone(),
+            vec![valid_interior],
+        )
+        .insert(&mut core)]);
```

Comment:
Can probably be shortened by using available APIs. Maybe (attention, pseudo-code) `Region::circle().add_interior(invalid_circle)`.